bentoml/_internal/service/inference_api.py

```diff
@@ -39,7 +45,77 @@ def __init__(
         InferenceAPI._validate_route(route)
 
         self.name = name
+        self.needs_ctx = False
+        self.ctx_param = None
         self.func = user_defined_callback
+        input_type = input_descriptor.input_type()
+        self.multi_input = isinstance(input_type, dict)
+        sig = inspect.signature(user_defined_callback)
+        if isinstance(input_type, dict):
+            # note: in python 3.6 kwarg order was not guaranteed to be preserved,
+            #       though it is in practice.
+            for key in input_type:
+                if key not in sig.parameters:
+                    if (
+                        key in ["context", "ctx"]
```

Comment:
Should we allow both `context` and `ctx`? Or allow only one.

Reply:
I'd want to lean towards allowing any name that could reasonably imply that a user wants a context parameter; but maybe we should just use any name left over after checking the rest of the IO descriptor defined ones?

Reply:
It probably doesn't provide much more flexibility for allowing any names. A lot of times, a well defined rule and naming is simpler for understanding.